cirq-core/cirq/study/resolver.py

```diff
@@ -63,7 +63,9 @@ def __init__(self, param_dict: 'cirq.ParamResolverOrSimilarType' = None) -> None
             return  # Already initialized. Got wrapped as part of the __new__.
 
         self._param_hash: Optional[int] = None
-        self.param_dict = cast(ParamDictType, {} if param_dict is None else param_dict)
+        self.param_dict = cast(
```

Comment:
why not rename `self.param_dict` -> `self._param_dict` and require access through `value_of` or some other read-only property. This--of course-- is a breaking change. We can define a `param_dict` property that raises a deprecation warning